cloudsync/tasks.py

```diff
@@ -209,3 +295,21 @@ def parse_content_metadata(response):
         content_length = int(content_length)
 
     return file_name, content_type, content_length
+
+
+def extract_title(filename):
+    """
+    Parses the filename for required course information
+    Args:
+        filename(str): The name of the video file, in format
+        'MIT-<course#>-<year>-<semester>-lec-mit-0000-<recording_date>-<time>-<session>.mp4'
+
+    Returns:
+        A string which should match a collection title: MIT-<course>-<session>-<year>-<semester>
+    """
+    rx = r'([A-Z]+)-([\w\.\s]+)-((\d{4})-(Fall|Spring)-)?lec-mit-0000-(\w+)-(\d+)(-([L\d\-]+))?(_\d+)?.*\.\w'
```

Comment:
If the regex changes significantly, the function's code that determines what part of the regex match equals course, session, etc might also need to change.